src/test/java/com/smartdoor/project/sensors/FaceRecogScannerTest.java

```diff
@@ -0,0 +1,178 @@
+        assertEquals(cameraParam.cameraAngle, updatedCameraParam.cameraAngle);
+    }
+
+    @Test
+    void getFeatureSet_ValidPhoto_FeatureSetReturned() throws Exception {
+        // Arrange
+        String filePath = "src/main/java/com/smartdoor/data/PhotoFeatureMap.json";
+        FileWriter file = new FileWriter(filePath);
+        file.write("{\"user3Photo\": {\"value\": \"1100\", \"lightIntensity\": 70, \"cameraAngle\": 35}}");
+        file.close();
+
+        Photo photo = new Photo();
+        photo.value = "user3Photo";
+
+        // Act
+        FeatureSet featureSet = faceRecogScanner.getFeatureSet(photo);
+
+        // Assert
+        assertNotNull(featureSet);
+        assertEquals("1100", featureSet.value);
+
+        // Clean up
+        File testFile = new File(filePath);
+        testFile.delete();
+    }
+
+    @Test
+    void getFeatureSet_InvalidPhoto_ExceptionThrown() {
+        // Arrange
+        Photo photo = new Photo();
+        photo.value = "invalidPhoto";
+
+        // Act & Assert
+        assertThrows(Exception.class, () -> {
+            faceRecogScanner.getFeatureSet(photo);
+        });
+    }
+
+    @Test
+    void getFeedback_LightIntensityLessThanThreshold_IncrementLightIntensity() {
+        // Arrange
+        CameraParam updatedCameraParam = new CameraParam();
+        updatedCameraParam.lightIntensity = 70;
+        Photo photo = new Photo();
+        photo.distance = 40;
+        Feedback feedback = new Feedback();
+
+        // Act
+        boolean validPhoto = faceRecogScanner.getFeedback(updatedCameraParam, photo, feedback, false);
+
+        // Assert
+        assertFalse(validPhoto);
+        assertEquals(80, updatedCameraParam.lightIntensity);
+    }
+
+    @Test
+    void getFeedback_CameraAngleLessThanThreshold_IncrementCameraAngle() {
+        // Arrange
+        CameraParam updatedCameraParam = new CameraParam();
+        updatedCameraParam.cameraAngle = 40;
+        Photo photo = new Photo();
+        photo.distance = 40;
+        Feedback feedback = new Feedback();
+
+        // Act
+        boolean validPhoto = faceRecogScanner.getFeedback(updatedCameraParam, photo, feedback, false);
+
+        // Assert
+        assertFalse(validPhoto);
+        assertEquals(50, updatedCameraParam.cameraAngle);
+    }
+
+    @Test
+    void getFeedback_DistanceGreaterThanThreshold_DecrementDistance() {
+        // Arrange
+        CameraParam updatedCameraParam = new CameraParam();
+        updatedCameraParam.cameraAngle = 40;
+        Photo photo = new Photo();
+        photo.distance = 60;
+        Feedback feedback = new Feedback();
+
+        // Act
+        boolean validPhoto = faceRecogScanner.getFeedback(updatedCameraParam, photo, feedback, false);
+
+        // Assert
+        assertFalse(validPhoto);
+        assertEquals(50, photo.distance);
+    }
+
+    @Test
+    void faceRecogProcessor_Captured_ReturnsFeatureSet() throws Exception {
+        // Arrange
+        boolean captured = true;
```

Comment:
@rnolas96 Change to `boolean captured = false` as code execution is not entering while loop.

And in `PhotoFeatureMap.json` the JSON single quotes are causing tests to fail. Change it to `"user3Photo": {"value": "1100", "lightIntensity": 10, "cameraAngle": 85}`